### PR TITLE
Fix chest opening bug

### PR DESCRIPTION
**Pull request by thienphuochoang**

### CODE REVIEW

**Review by Kanika834 — APPROVED** [review]

**Review by NoahMKuzyk — CHANGES_REQUESTED** [review]
Remove Main Menu.cs from commit; accident

**Review by Betadrone — APPROVED** [review]

**Review by NoahMKuzyk — APPROVED** [review]
Fixed, thx

File: CapstoneProject/Assets/Scripts/Core/Entity/Objects/Item.cs [hunk 1]
```diff
@@ -17,19 +21,18 @@ private void Start()
...
+            if (haveBeenInteracted || _player.GetComponent<KeyItemComponent>().TryToUseKeys() == false) return;
             animator.SetBool("Idle", false);
             animator.SetTrigger("Open");
             haveBeenInteracted = true;
-        }
-
-        private void OnTriggerStay2D(Collider2D other)
-        {
-            
+            _player.GetComponent<KeyItemComponent>().TryToUseKeys();
```
Comment (DoopweeSLC) [inline CapstoneProject/Assets/Scripts/Core/Entity/Objects/Item.cs:33]: try use keys returns a bool. if they don't have a key right now you still give them the coins and open? but if they have a key you take it away?

Comment (DoopweeSLC) [reply CapstoneProject/Assets/Scripts/Core/Entity/Objects/Item.cs:33]: I see you do this above in your check though, so if they have 2 keys, you'd use both on the one chest :(

**Pull request merged**